wger/exercises/api/views.py

```diff
@@ -57,12 +60,15 @@ class ExerciseViewSet(viewsets.ModelViewSet):
                      'license',
                      'license_author')
 
-    def pre_save(self, obj):
+    def perform_create(self, serializer):
         '''
-        Set language, author and status
+        Set author and status
         '''
-        obj.language = load_language()
+        language = load_language()
+        obj = serializer.save(language=language)
+        # Todo is it right to call set author after save?
```

Comment:
Yes, it should not be a problem. These only set the status and send an email. There are probably better ways to solve this, for example using signals, but then the code has no access to the request, or at least it wasn't the last time I looked at this.